static/src/javascripts/projects/commercial/modules/manage-ad-free-cookie-on-consent-change.ts

```diff
@@ -0,0 +1,29 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import { once } from 'lodash-es';
+import {
+	AdFreeCookieReasons,
+	maybeUnsetAdFreeCookie,
+	setAdFreeCookie,
+} from 'lib/manage-ad-free-cookie';
+import { removeSlots } from './remove-slots';
+
+/**
+ * If consent changes so that targeted advertising is disabled in tcfv2 regions,
+ * remove ad slots and set adFree cookie so the server won't render them on subsequent page loads
+ * Otherwise if in tcfv2 but consent to targeting is later allowed, removes the adFree cookie
+ */
+const manageAdFreeCookieOnConsentChange = once((): Promise<void> => {
+	onConsentChange((consent) => {
+		if (consent.tcfv2) {
+			if (!consent.canTarget) {
+				setAdFreeCookie(AdFreeCookieReasons.ConsentOptOut);
```

Comment:
Is it worth adding a unit test for this?

I think you could copy quite a bit of `initial-consent-state.spec.ts`

Reply:
good point!